src/sentry/api/endpoints/relay/project_configs.py

```diff
@@ -72,6 +67,47 @@ def _post(self, request: Request):
         else:
             return Response("Unsupported version, we only support versions 1 to 3.", 400)
 
+    def _should_use_v3(self, request):
+        version = request.GET.get("version")
+        set_tag("relay_endpoint_version", version)
+        no_cache = request.relay_request_data.get("noCache") or False
+        set_tag("relay_no_cache", no_cache)
+        is_full_config = request.relay_request_data.get("fullConfig")
+        set_tag("relay_full_config", is_full_config)
+
+        use_v3 = True
+        reject_reason = None
+
+        if version != "3":
+            use_v3 = False
+            reject_reason = "version"
+        elif not is_full_config:
+            # The v3 implementation can't handle partial configs. Relay by
+            # default request full configs and the amount of partial configs
+            # should be low, so we handle them per request instead of
+            # considering them v3.
+            use_v3 = False
+            reject_reason = "fullConfig"
+        elif no_cache:
+            use_v3 = False
+            reject_reason = "noCache"
+        elif random.random() >= options.get("relay.project-config-v3-enable"):
+            set_tag("relay_v3_sampled", False)
+            use_v3 = False
+            reject_reason = "sampling"
+        else:
+            set_tag("relay_v3_sampled", True)
+
+        set_tag("relay_use_v3", use_v3)
+        set_tag("relay_use_v3_rejected", reject_reason)
+        metrics.incr(
+            "api.endpoints.relay.project_configs._post",
```

Comment:
nit, I don't think we use underscores in metric names usually.

Reply:
We do use underscores, see
https://github.com/getsentry/sentry/blob/aa29ecdf772151846009e04ce090c875f44d80ae/src/sentry/api/endpoints/relay/project_configs.py#L87-L90

I tried to use the same name format (`project_configs`). I'm fine with not using underscores, but then we should update other metric names too. If we do that, I'll open a separate PR, since it's out of the scope of the present PR.

Reply:
This remains nothing more than a nit so do with it as you will.  I probably should have said "leading underscores".  Usually leading underscores have some sort of meaning by convention, and I wouldn't know what that is for metric names, hence this name looks a bit odd to me.